reference/5-api-publisher/readme.md

```diff
@@ -1,9 +1,130 @@
+
+### Configure Local Sandbox Environment
+
+Before using the API Publisher on a target ODS, you must create and configure an API client with the appropriate permissions for publishing.  The source and destination ODS/API must be of the same version.
+
+For Ed-Fi ODS / API 5.1 through 5.3 only: create and assign a claim set for the API Publisher by running the following database scripts:
+  * [Create-API-Publisher-Writer-Security-Metadata.sql](https://github.com/Ed-Fi-Alliance-OSS/Ed-Fi-API-Publisher/blob/main/eng/Create-API-Publisher-Writer-Security-Metadata.sql)
+  * [Configure-Minimal-Sandbox-Client-as-API-Publisher-Writer.sql](https://github.com/Ed-Fi-Alliance-OSS/Ed-Fi-API-Publisher/blob/main/eng/Configure-Minimal-Sandbox-Client-as-API-Publisher-Writer.sql)
+
+### Use the API Publisher
+
+The API Publisher has three options to use the product.  The API Publisher requires [.NET 8.0](https://dotnet.microsoft.com/en-us/download/dotnet/8.0) to run:
+
+#### Option 1 - From binaries
+
+ 1. Download the latest published API Publisher package here:  [Ed-Fi API Publisher v1.0](https://dev.azure.com/ed-fi-alliance/Ed-Fi-Alliance-OSS/_artifacts/feed/EdFi/NuGet/EdFi.ApiPublisher/overview/1.1.0).  Visit the page and click download.
+ 2. This will download a NuGet package to your computer.  Rename this file, `EdFi.ApiPublisher.1.1.0.nupkg`, to include .zip extension: `EdFi.ApiPublisher.1.1.0.zip`.
+ 3. The binary mentioned below is in the `EdFi.ApiPublisher.Win64` folder, as `EdFiApiPublisher.exe`.
+
+#### Option 2 - From Docker img
+
+The Docker image for the Ed-Fi API Publisher is available here: [Ed-Fi API Publisher v1.0 on Docker Hub](https://hub.docker.com/layers/edfialliance/ods-api-publisher/v1.1.0/images/sha256-23a6989b16d779d594692e1c9a8f2eac1790ed53312d424d85e869fec05eb914?context=explore).  Use this to include in your Docker environment and alongside other components of the Ed-Fi stack.
+
+#### Option 3 - Build the API Publisher from source code
+
+If you would like to build the API Publisher from source, build the solution by running the following command from the repository's root directory:
+
+`dotnet build`
+
+The API Publisher executable (`EdFiApiPublisher.exe`) will be located in the _.\EdFi.Tools.ApiPublisher.Cli\bin\Debug\net8.0_ subfolder.
+
+### Publish Data to Local Sandbox
+
+> **IMPORTANT: After changing security metadata for the API, YOU MUST RESTART the local sandbox Ed-Fi ODS API if it is already running.**
+
+Next, locate the key/secret for the API client for the minimal template sandbox. You can use the Sandbox Admin tool, or can just run the following query against the `EdFi_Admin` database:
+
+```sql
+SELECT  [Key], [Secret]
+FROM    EdFi_Admin.dbo.ApiClients
+WHERE   Name = 'Minimal Demonstration Sandbox'
+```
```

Comment:
As soon as someone uses that key and secret in Swagger, the `secret` value will become encrypted. So this command won't work at that point. If using the Sandbox Admin tool, instead direct the end-user to find the default key and secret in the appsettings file.

Reply:
query removed and comment updated